schema/elasticsearch/visibility/versioned/v3/upgrade.sh

```diff
@@ -0,0 +1,59 @@
+#!/usr/bin/env bash
+
+set -eu -o pipefail
+
+# Prerequisites:
+#   - jq
+#   - curl
+
+# Input parameters.
+: "${ES_SCHEME:=http}"
+: "${ES_SERVER:=127.0.0.1}"
+: "${ES_PORT:=9200}"
+: "${ES_USER:=}"
+: "${ES_PWD:=}"
+: "${ES_VERSION:=v7}"
+: "${ES_VIS_INDEX_V1:=temporal_visibility_v1_dev}"
+: "${AUTO_CONFIRM:=}"
+: "${SLICES_COUNT:=auto}"
+
+es_endpoint="${ES_SCHEME}://${ES_SERVER}:${ES_PORT}"
+
+echo "=== Step 0. Sanity check if Elasticsearch index is accessible ==="
+
+if ! curl --silent --fail --user "${ES_USER}":"${ES_PWD}" "${es_endpoint}/${ES_VIS_INDEX_V1}/_stats/docs" --write-out "\n"; then
+    echo "Elasticsearch index ${ES_VIS_INDEX_V1} is not accessible at ${es_endpoint}."
+    exit 1
+fi
+
+echo "=== Step 1. Add new builtin search attributes ==="
+
+case $ES_VERSION in
+    v6) date_type='date'       ; doc_type='/_doc' ;;
+    *)  date_type='date_nanos' ; doc_type=''      ;;
+esac
+
+new_mapping='
+{
+  "properties": {
+    "TemporalNamespaceDivision": {
```

Comment:
This is getting kind of annoying. What do you think about auto-registering new pre-defined on upgrade? Or the cli tool?

Reply:
We are starting to get ES schema changes more often and expectation was the opposite. Ideally, special ES CLI tool needs to be added (there is an open #2977 PR for it) which will handle all ES schema updates. I would leave it as is for now.